Switcher and Bottom Tabs Animations and UI Performance Improvements

Parveshdhull opened the pull request:
fixes https://github.com/status-im/status-react/issues/13452
fixes https://github.com/status-im/status-react/issues/13440

### Summary

- Implemented Reanimated V2
- Migrated Switcher animations to Reanimated V2
- Added bottom tabs & Stacks Animations
- Improved bottom tabs, tab changing performance
- Polished android & IOS UI

### Testing
Please test PR for any regressions.
Also, PR includes a new source folder(js) for `worklets`, please let me know if this is creating an issue for the nightly build. (No need for retesting, just let me know if the nightly build is failing)

status: ready

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 38833fdb | [#1](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-13470/1/) | 2022-06-10 00:20:29 | ~6 min | `android-e2e` | [:page_facing_up:`log`](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-13470/1/consoleText) |

Parveshdhull commented:
```
[2022-06-10T00:20:26.493Z] Error: Unable to resolve module `../js/worklet_factory.js` from `app/index.js`: 
[2022-06-10T00:20:26.493Z] 
[2022-06-10T00:20:26.493Z] None of these files exist:
[2022-06-10T00:20:26.493Z]   * js/worklet_factory.js(.native|.android.js|.native.js|.js|.android.json|.native.json|.json|.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.tsx)
```
Not sure why the file is not found on the Jenkins build, working great locally.

Parveshdhull commented on src/status_im/switcher/switcher.cljs:
```diff
@@ -1,44 +1,64 @@
+           switcher-button-animated-style (reanimated/apply-animations-to-style
+                                           {:opacity switcher-button-opacity}
+                                           switcher-button-original-style)]
+       (reagent/as-element
```
I liked the reagent's shorthand `[:f>` for creating [functional components](https://reactjs.org/docs/components-and-props.html#function-and-class-components), but for some reason it is not working, (only in android PR build) and giving runtime error `TypeError: Cannot read property 'indexOf' of null`. 
So using this `[:> (fn [] ... (reagent/as-element` method for now.

Not only for `[:f>`, if I use any non existing shorthand like `[:x>`, it still gives same error. But in local build if I use non existing shorthand it gives error, non matching tag used. So it looks like it is not understanding these shorthand's in android PR build.

Any thought, why that might be happening?

flexsurfer replied:
could you provide an example ? 

also you could try to set
```
:static-fns true
:fn-invoke-direct true
```

Parveshdhull replied:
> could you provide an example ?

```
(defn child-comp []
  [:f> (fn []
         [react/view])])
```

If I use above child comp inside any parent component, then the PR build app crashes at runtime with `TypeError: Cannot read property 'indexOf' of null`, but If I use

```
(defn child-comp []
  [:> (fn []
        (reagent/as-element
         [react/view]))])
```

Then it just works, not sure why first one is failing

> also you could try to set
> ```
> :static-fns true
> :fn-invoke-direct true
> ```

it is already there, https://github.com/status-im/status-react/blob/9f9b896eb7fcfac60d55b33a3744480b13d2ba8d/shadow-cljs.edn#L64

flexsurfer replied:
i mean in :dev build. to see if its a reason

Parveshdhull replied:
**Fixed**: _Adding solution for reference_

CI build was using a different version of the reagent (0.10.0), than the dev build(1.0.0). And `:f>` is only available in `1.0.0+` versions.
`make nix-update-clojure` fixed the issue.

Thank you @flexsurfer and @Jakubgs

Parveshdhull commented on .clj-kondo/config.edn:
```diff
@@ -1,6 +1,6 @@
 {:lint-as {status-im.utils.views/defview  clojure.core/defn
            status-im.utils.views/letsubs  clojure.core/let
-           reagent.core/with-let          clojkure.core/let
+           reagent.core/with-let          clojure.core/let
```
typo?

Parveshdhull commented:
I will work on polishing UI and animations in the same PR because merging in the current state feels a little broken. I will ping for testing and review later once PR will be ready.

du64 commented:
Any plans to speed the animations up? The one for the switcher is quite slow, which kinda removes the purpose of switching quickly between things

Parveshdhull commented:
> Any plans to speed the animations up? The one for the switcher is quite slow, which kinda removes the purpose of switching quickly between things

Hi @du64, Thank you very much for the feedback and checking out the PR.

Currently, the switcher is still in the experimentation/development phase, and we don't have a clear picture of its performance with all switcher cards.
Once everything will be implemented, we will have better idea.
And then we can easily tweak animation duration(currently 300 ms) and can use whichever value gives best optimum experience. Thank you

flexsurfer commented on src/status_im/switcher/switcher_container.cljs:
```diff
@@ -28,7 +28,5 @@
-  [rn/view {:style (styles/switcher-screen-container
-                    view-id switcher-screen-radius)}
-   [switch-screen toggle-switcher-screen]])
+(defn tabs [toggle-switcher-screen]
```
why tabs is needed , why not `switch-screen` directly?

Parveshdhull replied:
Hi, currently we have only a switcher screen in the tab, that's why
  ```
  (defn tabs [toggle-switcher-screen]
    [switch-screen toggle-switcher-screen])
  ```
  But it is more like a placeholder for now, once the swipable tabs and other screens are implemented it will be something like
  ```
  (defn tabs [toggle-switcher-screen]
    [swipeable-tabs
      [switch-screen toggle-switcher-screen]
      [share-screen]
      [scan-center-screen]
      [activity-center-screen]])
  ```

flexsurfer commented on src/status_im/switcher/cards/messaging_card.cljs:
```diff
@@ -7,7 +7,7 @@
 
 (defn on-press [id toggle-switcher-screen]
-  (toggle-switcher-screen)
+  (js/setTimeout toggle-switcher-screen 100)
```
could you add a comment pls

flexsurfer commented on src/quo2/reanimated.cljs:
```diff
@@ -0,0 +1,69 @@
+       (into {})))
+
+;; Worklets
+(def worklet-factory (js/require "../js/worklet_factory.js"))
```
can it be in `:require` ? `["../js/worklet_factory.js" :as worklet-factory]`

flexsurfer replied:
btw why it's not inside src folder?

Parveshdhull replied:
> can it be in :require ? ["../js/worklet_factory.js" :as worklet-factory]

js directive are not working with :require https://github.com/status-im/status-react/issues/13452#issuecomment-1150540973

> btw why it's not inside src folder?

moved to src

flexsurfer commented on src/quo2/reanimated.cljs:
```diff
@@ -0,0 +1,69 @@
+
+;; Animators
+(defn animate-shared-value-with-timing [anim val duration easing]
+  (set-shared-value anim (with-timing val (clj->js {:duration duration
```
clj->js is slow, probably better to use `(js-obj "foo" 1 "bar" 2)` here

flexsurfer commented on src/quo2/reanimated.cljs:
```diff
@@ -0,0 +1,69 @@
+;; kebab-case styles are not working for worklets
+;; so first convert kebab case styles into camel case styles
+(defn apply-animations-to-style [animations style]
+  (let [animations (map-keys kebab-case->camelCase animations)
```
could you elaborate why map-keys and kebab-case->camelCase

Parveshdhull replied:
Hi, I think kebab case only works with clojure due to reagent, but as worklets are inside js file which is directly processed by metro instead of shadow-cljs (due to js/require for directives) only camel case works there

flexsurfer commented on src/status_im/switcher/constants.cljs:
```diff
@@ -2,31 +2,51 @@
 (defn dimensions []
   (<sub [:dimensions/window]))
+
+(def tabs '("communities" "chats" "wallet" "browser"))
```
why not vector ?

flexsurfer commented on src/status_im/switcher/animation.cljs:
```diff
@@ -1,47 +1,67 @@
+
+
+(defn bottom-tab-on-press [shared-values selected-tab-id]
+  (let [selected-tab (get (string/split (name selected-tab-id) #"-") 0)]
```
thats strange to do it each time on press, better to have a map ?

flexsurfer commented on src/status_im/switcher/animation.cljs:
```diff
@@ -1,47 +1,67 @@
+  (let [selected-tab (get (string/split (name selected-tab-id) #"-") 0)]
+    (doseq [tab constants/tabs]
+      (let [selected-tab?              (= tab selected-tab)
+            tab-opacity-shared-value   (get shared-values (keyword (str tab "-tab-opacity")))
```
same here

flexsurfer commented on src/status_im/switcher/bottom_tabs.cljs:
```diff
@@ -1,30 +1,68 @@
+
+(defn tabs [shared-values icons-only?]
+  [rn/view {:style (styles/bottom-tabs icons-only?)}
+   [bottom-tab :main-icons/redesign-communities "communities" icons-only? shared-values]
```
why do you need string name ?

flexsurfer replied:
why not keyword ?

Parveshdhull replied:
oh I missed this change, doing now

Parveshdhull replied:
done

flexsurfer commented on src/status_im/switcher/bottom_tabs.cljs:
```diff
@@ -1,30 +1,68 @@
-                       :bottom-tabs-non-selected-tab))]])
+(defn bottom-tab [icon tab icons-only? shared-values]
+  [:>
+   (fn []
```
could you elaborate why `[:> (fn []`

flexsurfer commented on src/status_im/switcher/home_stack.cljs:
```diff
@@ -0,0 +1,67 @@
+(defn home []
+  [:>
+   (fn []
+     (let [selected-tab   (get (string/split (name @bottom-tabs/selected-tab-id) #"-") 0)
```
`string/split (name` doesn't look good , probably there is a better way

flexsurfer commented on nix/mobile/android/release.nix:
```diff
@@ -52,7 +52,7 @@ in stdenv.mkDerivation rec {
       include = [
         "package.json" "yarn.lock" "metro.config.js"
-        "resources/.*" "translations/.*"
+        "resources/.*" "translations/.*" "src/js/.*" ".babelrc"
```
but src should be included , no ?

jakubgs replied:
No, it should not be necessary, since this is the Android Gradle build.
Are we sure this change shouldn't go to `nix/mobile/android/jsbundle/default.nix`?

Parveshdhull replied:
Hi, @jakubgs `jsbundle/default.nix` already has `"src/.*"` but android build still complains about file not found. It only works when added to 'release.nix`

jakubgs replied:
What file does it complain about exactly?

Parveshdhull replied:
So for using JavaScript Directives for creating [reanimated worklets](https://stackoverflow.com/questions/72500152/how-to-use-reanimated-worklet-directive-in-clojurescript/72513996) in the clojure project, I needed to use the js file.

For that, I created a new file `src/js/worklet-factory.js`. But when the android app is built it complains that this [file is not found](https://github.com/status-im/status-react/pull/13470#issuecomment-1151745897). 

This file is in the `"src/.*"` directory which we bundle in `nix/mobile/android/jsbundle/default.nix`, still it complains. Maybe because of how it is [imported](https://github.com/status-im/status-react/pull/13470#discussion_r907148059), this file acts more like a separate module.

jakubgs replied:
Then if only `src/js/worklet-factory.js` is necessary, why are you adding all of `src`?

flexsurfer commented on src/mocks/js_dependencies.cljs:
```diff
@@ -308,7 +308,7 @@
     "@react-native-community/async-storage" async-storage
     "react-native-svg" react-native-svg
-    "../js/worklet_factory.js" worklet-factory
+    "../src/js/worklet_factory.js" worklet-factory
```
./js/worklet_factory.js ?

flexsurfer commented on src/quo2/reanimated.cljs:
```diff
@@ -47,7 +47,7 @@
 
 ;; Worklets
-(def worklet-factory (js/require "../js/worklet_factory.js"))
+(def worklet-factory (js/require "../src/js/worklet_factory.js"))
```
./js/worklet_factory.js ?

Parveshdhull replied:
For some reason `js/require` is only working with `../src/js/worklet_factory.js`

"https://user-images.githubusercontent.com/17097240/175988135-909b2ee2-a4fe-4ad5-ae84-1aaf30bdc953.mp4"

> Github video player is unable to play this video, only playing locally

flexsurfer reviewed (approved):
great job!

Parveshdhull commented:
Hi @qoqobolo, Thanks for testing the PR and finding the issue.
Issue should be fixed now.

status-im-auto commented:
## 100% of end-end tests have passed
```
Total executed tests: 84
Failed tests: 0
Passed tests: 84
```
<h3>Passed tests (84)</h3><details><summary>Click to expand</summary><br/><br/><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892682">test_pairing_sync_1_1_chat_message</a>, id: 702198 </b></td></tr><tr><td><p><b>Class: TestPairingSyncMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f8d70ad9fc9c4d8bb2593edbd22afaa2?auth=cd8cb013e81a4cbf531036502f14b536#431">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/708540b1ae634f81a52d5d5d8c8bb1f6?auth=22cf2fd1d6a5ff81b9a821b9b10b1c7a#131">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892676">test_cellular_settings_on_off_public_chat_fetching_history</a>, id: 702188 </b></td></tr><tr><td><p><b>Class: TestContactBlockMigrateKeycardMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/127b86273aee43e1ba5346611f788fbf?auth=61ca8fd37b367111dfddebe8ef8f5f29#492">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f857ee7768c84fb6abeefe2acdb8f8b3?auth=0a736679c4267cb69211ab5b19971947#287">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892669">test_ens_mention_push_highlighted_public_chat</a>, id: 702156 </b></td></tr><tr><td><p><b>Class: TestEnsStickersMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/81c5c5a6ce5a494ba5a4bc9edc8c1490?auth=209d6ac819376ad35cc943d127c7196c#246">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/86ec612fa9554c4faef301569d7949cd?auth=f518895eb9cfbb12160ae49118694cf6#184">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892678">test_pairing_sync_initial_contacts_blocked_users</a>, id: 702194 </b></td></tr><tr><td><p><b>Class: TestPairingSyncMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f8d70ad9fc9c4d8bb2593edbd22afaa2?auth=cd8cb013e81a4cbf531036502f14b536#277">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/708540b1ae634f81a52d5d5d8c8bb1f6?auth=22cf2fd1d6a5ff81b9a821b9b10b1c7a#69">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892629">test_public_chat_message_edit</a>, id: 700734 </b></td></tr><tr><td><p><b>Class: TestPublicChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f4fd24d3b9bd4cdba84d51440ecc3eb5?auth=62d5d0fddaf4087a0f8e831b9a23fcde#119">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/60f9247c10834c278be8564228c98d5b?auth=d55103e69af7a153dbb65a3451527815#77">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892633">test_public_chat_tag_message</a>, id: 700738 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#203">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892643">test_restore_seed_phrase_field_validation</a>, id: 700750 </b></td></tr><tr><td><p><b>Class: TestRestoreOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/91b738e95d584fbb8e3e8e44d877ade9?auth=4ca47f340bfaa17fd8c8cd05ba132451#167">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892662">test_browser_offline</a>, id: 702075 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#365">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892640">test_onboarding_add_new_multiaccount_username_by_position_pass_validation</a>, id: 700747 </b></td></tr><tr><td><p><b>Class: TestOnboardingOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/abfad0c25a9f4690b5462cc823deab33?auth=7cfcae6719fa774c73323d8a288ff9a6#378">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892606">test_public_chat_copy_and_paste_message_in_chat_input</a>, id: 5317 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#164">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>11. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892672">test_contact_add_remove_mention_default_username_nickname_public_chat</a>, id: 702175 </b></td></tr><tr><td><p><b>Class: TestContactBlockMigrateKeycardMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/127b86273aee43e1ba5346611f788fbf?auth=61ca8fd37b367111dfddebe8ef8f5f29#223">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f857ee7768c84fb6abeefe2acdb8f8b3?auth=0a736679c4267cb69211ab5b19971947#233">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>12. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892681">test_pairing_sync_contacts_add_remove_set_nickname_ens</a>, id: 702197 </b></td></tr><tr><td><p><b>Class: TestPairingSyncMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f8d70ad9fc9c4d8bb2593edbd22afaa2?auth=cd8cb013e81a4cbf531036502f14b536#355">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/708540b1ae634f81a52d5d5d8c8bb1f6?auth=22cf2fd1d6a5ff81b9a821b9b10b1c7a#110">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>13. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892637">test_onboarding_share_wallet_address</a>, id: 700744 </b></td></tr><tr><td><p><b>Class: TestOnboardingOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/abfad0c25a9f4690b5462cc823deab33?auth=7cfcae6719fa774c73323d8a288ff9a6#170">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>14. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892653">test_send_tx_custom_token_18_decimals_invalid_password</a>, id: 700765 </b></td></tr><tr><td><p><b>Class: TestSendTxDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/76a39fc9243143538e19cae6114a3e9b?auth=87e0b0236541f1e45d3d45e6bd6b6703#478">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>15. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892614">test_keycard_create_account_unlock_same_seed</a>, id: 5689 </b></td></tr><tr><td><p><b>Class: TestKeycardTxOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/925aed991abc41c7a610d918d992a1be?auth=5e4eb8ef6d344f1849d6d949aa31411e#647">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>16. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892680">test_pairing_sync_contacts_block_unblock</a>, id: 702196 </b></td></tr><tr><td><p><b>Class: TestPairingSyncMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f8d70ad9fc9c4d8bb2593edbd22afaa2?auth=cd8cb013e81a4cbf531036502f14b536#287">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/708540b1ae634f81a52d5d5d8c8bb1f6?auth=22cf2fd1d6a5ff81b9a821b9b10b1c7a#96">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>17. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892609">test_public_chat_unread_messages_counter</a>, id: 5360 </b></td></tr><tr><td><p><b>Class: TestPublicChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f4fd24d3b9bd4cdba84d51440ecc3eb5?auth=62d5d0fddaf4087a0f8e831b9a23fcde#172">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/60f9247c10834c278be8564228c98d5b?auth=d55103e69af7a153dbb65a3451527815#103">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>18. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892625">test_public_chat_unread_messages_counter_for_mention_relogin</a>, id: 700718 </b></td></tr><tr><td><p><b>Class: TestPublicChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f4fd24d3b9bd4cdba84d51440ecc3eb5?auth=62d5d0fddaf4087a0f8e831b9a23fcde#197">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/60f9247c10834c278be8564228c98d5b?auth=d55103e69af7a153dbb65a3451527815#111">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>19. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892630">test_public_chat_message_delete</a>, id: 700735 </b></td></tr><tr><td><p><b>Class: TestPublicChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f4fd24d3b9bd4cdba84d51440ecc3eb5?auth=62d5d0fddaf4087a0f8e831b9a23fcde#134">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/60f9247c10834c278be8564228c98d5b?auth=d55103e69af7a153dbb65a3451527815#86">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>20. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892657">test_keycard_send_tx_sign_message_request_stt_testdapp</a>, id: 700769 </b></td></tr><tr><td><p><b>Class: TestKeycardTxOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/925aed991abc41c7a610d918d992a1be?auth=5e4eb8ef6d344f1849d6d949aa31411e#293">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>21. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892621">test_1_1_chat_image_send_save_reply</a>, id: 6305 </b></td></tr><tr><td><p><b>Class: TestOneToOneChatMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cbd7bd3ff412482298a365d09f8c8829?auth=3adcf523ed78b978deef136e6c2114bb#309">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ec00402e11d8496eac569953c24c3ac0?auth=c94915bfebf44dcc6e213a63fa5806ee#265">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>22. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892655">test_keycard_send_tx_eth</a>, id: 700767 </b></td></tr><tr><td><p><b>Class: TestKeycardTxOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/925aed991abc41c7a610d918d992a1be?auth=5e4eb8ef6d344f1849d6d949aa31411e#146">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>23. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892667">test_ens_command_send_tx_eth_1_1_chat</a>, id: 702153 </b></td></tr><tr><td><p><b>Class: TestEnsStickersMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/81c5c5a6ce5a494ba5a4bc9edc8c1490?auth=209d6ac819376ad35cc943d127c7196c#92">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/86ec612fa9554c4faef301569d7949cd?auth=f518895eb9cfbb12160ae49118694cf6#151">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>24. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892659">test_browser_blocked_url</a>, id: 702072 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#250">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>25. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892649">test_wallet_add_hide_unhide_account_private_key</a>, id: 700761 </b></td></tr><tr><td><p><b>Class: TestWalletManagementDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f41b5bca6c3f45208d5c5b2bb0ab5b2b?auth=fd9f99e71b120835b46835a337bcd92c#467">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>26. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892611">test_1_1_chat_delete_via_delete_button_relogin</a>, id: 5387 </b></td></tr><tr><td><p><b>Class: TestOneToOneChatMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cbd7bd3ff412482298a365d09f8c8829?auth=3adcf523ed78b978deef136e6c2114bb#438">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ec00402e11d8496eac569953c24c3ac0?auth=c94915bfebf44dcc6e213a63fa5806ee#394">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>27. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892636">test_onboarding_share_contact_address</a>, id: 700743 </b></td></tr><tr><td><p><b>Class: TestOnboardingOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/abfad0c25a9f4690b5462cc823deab33?auth=7cfcae6719fa774c73323d8a288ff9a6#98">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>28. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892602">test_group_chat_offline_pn</a>, id: 3998 </b></td></tr><tr><td><p><b>Class: TestGroupChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/fe12464090874b3e931b720d1684deed?auth=86365435160d9c916e48bd9c52ca6c80#173">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/9958600248224e8b82d4ea0895cca118?auth=f774c6cba0e6284c1902c9850c6c6208#127">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/75bbbe8203d741fa9ec8702f9aca66b3?auth=04d70d7bf1b4aa4013b20ace990df098#104">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>29. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892622">test_1_1_chat_message_reaction</a>, id: 6315 </b></td></tr><tr><td><p><b>Class: TestOneToOneChatMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cbd7bd3ff412482298a365d09f8c8829?auth=3adcf523ed78b978deef136e6c2114bb#71">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ec00402e11d8496eac569953c24c3ac0?auth=c94915bfebf44dcc6e213a63fa5806ee#64">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>30. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892603">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 5310 </b></td></tr><tr><td><p><b>Class: TestOneToOneChatMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cbd7bd3ff412482298a365d09f8c8829?auth=3adcf523ed78b978deef136e6c2114bb#372">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ec00402e11d8496eac569953c24c3ac0?auth=c94915bfebf44dcc6e213a63fa5806ee#337">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>31. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892661">test_browser_invalid_url</a>, id: 702074 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#351">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>32. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892607">test_public_chat_delete_chat_long_press</a>, id: 5319 </b></td></tr><tr><td><p><b>Class: TestPublicChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f4fd24d3b9bd4cdba84d51440ecc3eb5?auth=62d5d0fddaf4087a0f8e831b9a23fcde#229">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/60f9247c10834c278be8564228c98d5b?auth=d55103e69af7a153dbb65a3451527815#157">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>33. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892652">test_send_tx_token_7_decimals</a>, id: 700764 </b></td></tr><tr><td><p><b>Class: TestSendTxDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/76a39fc9243143538e19cae6114a3e9b?auth=87e0b0236541f1e45d3d45e6bd6b6703#262">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>34. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892650">test_wallet_add_account_seed_phrase_validation</a>, id: 700762 </b></td></tr><tr><td><p><b>Class: TestWalletManagementDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f41b5bca6c3f45208d5c5b2bb0ab5b2b?auth=fd9f99e71b120835b46835a337bcd92c#705">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>35. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892658">test_keycard_wallet_recover_pairing_check_balance_after_offline_tx_history</a>, id: 700770 </b></td></tr><tr><td><p><b>Class: TestKeycardTxOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/925aed991abc41c7a610d918d992a1be?auth=5e4eb8ef6d344f1849d6d949aa31411e#527">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>36. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892610">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 5373 </b></td></tr><tr><td><p><b>Class: TestOneToOneChatMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cbd7bd3ff412482298a365d09f8c8829?auth=3adcf523ed78b978deef136e6c2114bb#108">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ec00402e11d8496eac569953c24c3ac0?auth=c94915bfebf44dcc6e213a63fa5806ee#113">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>37. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892638">test_onboarding_backup_seed_phrase_restore_same_login_logcat</a>, id: 700745 </b></td></tr><tr><td><p><b>Class: TestOnboardingOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/abfad0c25a9f4690b5462cc823deab33?auth=7cfcae6719fa774c73323d8a288ff9a6#242">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>38. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892612">test_public_chat_navigate_to_chat_when_relaunch</a>, id: 5396 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#132">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>39. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892670">test_sticker_1_1_public_chat</a>, id: 702157 </b></td></tr><tr><td><p><b>Class: TestEnsStickersMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/81c5c5a6ce5a494ba5a4bc9edc8c1490?auth=209d6ac819376ad35cc943d127c7196c#272">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/86ec612fa9554c4faef301569d7949cd?auth=f518895eb9cfbb12160ae49118694cf6#221">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>40. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892664">test_browser_bookmarks_create_edit_remove</a>, id: 702077 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#488">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>41. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892620">test_1_1_chat_push_emoji</a>, id: 6283 </b></td></tr><tr><td><p><b>Class: TestOneToOneChatMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cbd7bd3ff412482298a365d09f8c8829?auth=3adcf523ed78b978deef136e6c2114bb#284">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ec00402e11d8496eac569953c24c3ac0?auth=c94915bfebf44dcc6e213a63fa5806ee#227">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>42. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892623">test_1_1_chat_text_message_with_push</a>, id: 6316 </b></td></tr><tr><td><p><b>Class: TestOneToOneChatMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cbd7bd3ff412482298a365d09f8c8829?auth=3adcf523ed78b978deef136e6c2114bb#90">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ec00402e11d8496eac569953c24c3ac0?auth=c94915bfebf44dcc6e213a63fa5806ee#83">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>43. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892642">test_restore_set_up_wallet_sign_phrase</a>, id: 700749 </b></td></tr><tr><td><p><b>Class: TestRestoreOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/91b738e95d584fbb8e3e8e44d877ade9?auth=4ca47f340bfaa17fd8c8cd05ba132451#66">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>44. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892671">test_start_new_chat_public_key_validation</a>, id: 702158 </b></td></tr><tr><td><p><b>Class: TestEnsStickersMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/81c5c5a6ce5a494ba5a4bc9edc8c1490?auth=209d6ac819376ad35cc943d127c7196c#344">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/86ec612fa9554c4faef301569d7949cd?auth=f518895eb9cfbb12160ae49118694cf6#360">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>45. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892677">test_restore_account_migrate_multiaccount_to_keycard_no_db_saved_add_wallet_send_tx</a>, id: 702189 </b></td></tr><tr><td><p><b>Class: TestRestoreOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/91b738e95d584fbb8e3e8e44d877ade9?auth=4ca47f340bfaa17fd8c8cd05ba132451#224">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>46. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892601">test_group_chat_leave_relogin</a>, id: 3997 </b></td></tr><tr><td><p><b>Class: TestGroupChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/fe12464090874b3e931b720d1684deed?auth=86365435160d9c916e48bd9c52ca6c80#250">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/9958600248224e8b82d4ea0895cca118?auth=f774c6cba0e6284c1902c9850c6c6208#179">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/75bbbe8203d741fa9ec8702f9aca66b3?auth=04d70d7bf1b4aa4013b20ace990df098#139">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>47. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892641">test_restore_uppercase_whitespaces_seed_phrase_special_char_passw_logcat</a>, id: 700748 </b></td></tr><tr><td><p><b>Class: TestRestoreOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/91b738e95d584fbb8e3e8e44d877ade9?auth=4ca47f340bfaa17fd8c8cd05ba132451#42">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>48. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892647">test_wallet_add_account_generate_new</a>, id: 700759 </b></td></tr><tr><td><p><b>Class: TestWalletManagementDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f41b5bca6c3f45208d5c5b2bb0ab5b2b?auth=fd9f99e71b120835b46835a337bcd92c#205">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>49. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892634">test_public_chat_open_using_deep_link</a>, id: 700739 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#232">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>50. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892648">test_wallet_add_delete_watch_only_account</a>, id: 700760 </b></td></tr><tr><td><p><b>Class: TestWalletManagementDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f41b5bca6c3f45208d5c5b2bb0ab5b2b?auth=fd9f99e71b120835b46835a337bcd92c#328">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>51. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892673">test_contact_block_unblock_public_chat_offline</a>, id: 702176 </b></td></tr><tr><td><p><b>Class: TestContactBlockMigrateKeycardMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/127b86273aee43e1ba5346611f788fbf?auth=61ca8fd37b367111dfddebe8ef8f5f29#352">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f857ee7768c84fb6abeefe2acdb8f8b3?auth=0a736679c4267cb69211ab5b19971947#240">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>52. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892604">test_public_chat_message_send_check_timestamps_while_on_different_tab</a>, id: 5313 </b></td></tr><tr><td><p><b>Class: TestPublicChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f4fd24d3b9bd4cdba84d51440ecc3eb5?auth=62d5d0fddaf4087a0f8e831b9a23fcde#101">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/60f9247c10834c278be8564228c98d5b?auth=d55103e69af7a153dbb65a3451527815#59">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>53. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892635">test_onboarding_home_initial_popup</a>, id: 700742 </b></td></tr><tr><td><p><b>Class: TestOnboardingOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/abfad0c25a9f4690b5462cc823deab33?auth=7cfcae6719fa774c73323d8a288ff9a6#60">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>54. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892605">test_1_1_chat_non_latin_message_to_newly_added_contact_with_profile_picture_on_different_networks</a>, id: 5315 </b></td></tr><tr><td><p><b>Class: TestOneToOneChatMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cbd7bd3ff412482298a365d09f8c8829?auth=3adcf523ed78b978deef136e6c2114bb#197">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ec00402e11d8496eac569953c24c3ac0?auth=c94915bfebf44dcc6e213a63fa5806ee#183">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>55. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892679">test_pairing_sync_initial_public_chats</a>, id: 702195 </b></td></tr><tr><td><p><b>Class: TestPairingSyncMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f8d70ad9fc9c4d8bb2593edbd22afaa2?auth=cd8cb013e81a4cbf531036502f14b536#280">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/708540b1ae634f81a52d5d5d8c8bb1f6?auth=22cf2fd1d6a5ff81b9a821b9b10b1c7a#87">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>56. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892668">test_ens_mention_nickname_1_1_chat</a>, id: 702155 </b></td></tr><tr><td><p><b>Class: TestEnsStickersMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/81c5c5a6ce5a494ba5a4bc9edc8c1490?auth=209d6ac819376ad35cc943d127c7196c#173">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/86ec612fa9554c4faef301569d7949cd?auth=f518895eb9cfbb12160ae49118694cf6#169">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>57. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892627">test_group_chat_join_send_text_messages_push</a>, id: 700731 </b></td></tr><tr><td><p><b>Class: TestGroupChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/fe12464090874b3e931b720d1684deed?auth=86365435160d9c916e48bd9c52ca6c80#133">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/9958600248224e8b82d4ea0895cca118?auth=f774c6cba0e6284c1902c9850c6c6208#98">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/75bbbe8203d741fa9ec8702f9aca66b3?auth=04d70d7bf1b4aa4013b20ace990df098#71">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>58. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892615">test_group_chat_highligted</a>, id: 5756 </b></td></tr><tr><td><p><b>Class: TestGroupChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/fe12464090874b3e931b720d1684deed?auth=86365435160d9c916e48bd9c52ca6c80#231">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/9958600248224e8b82d4ea0895cca118?auth=f774c6cba0e6284c1902c9850c6c6208#149">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/75bbbe8203d741fa9ec8702f9aca66b3?auth=04d70d7bf1b4aa4013b20ace990df098#132">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>59. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892639">test_onboarding_cant_sign_in_with_invalid_password_logcat</a>, id: 700746 </b></td></tr><tr><td><p><b>Class: TestOnboardingOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/abfad0c25a9f4690b5462cc823deab33?auth=7cfcae6719fa774c73323d8a288ff9a6#342">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>60. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892666">test_ens_purchased_in_profile</a>, id: 702152 </b></td></tr><tr><td><p><b>Class: TestEnsStickersMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/81c5c5a6ce5a494ba5a4bc9edc8c1490?auth=209d6ac819376ad35cc943d127c7196c#89">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/86ec612fa9554c4faef301569d7949cd?auth=f518895eb9cfbb12160ae49118694cf6#125">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>61. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892624">test_1_1_chat_text_message_edit_delete_push_disappear</a>, id: 695843 </b></td></tr><tr><td><p><b>Class: TestOneToOneChatMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/cbd7bd3ff412482298a365d09f8c8829?auth=3adcf523ed78b978deef136e6c2114bb#148">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/ec00402e11d8496eac569953c24c3ac0?auth=c94915bfebf44dcc6e213a63fa5806ee#136">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>62. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892617">test_1_1_chat_command_request_and_send_tx_stt_in_1_1_chat_offline</a>, id: 6263 </b></td></tr><tr><td><p><b>Class: TestCommandsMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/eceb4772387f42b0b085aba967b25cf0?auth=04ba18d8d3ff71837e69e9938b5ea326#217">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/de559aba346c48069881752c907a0c8c?auth=5adf98350d693108090df86c534ec14d#179">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>63. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892675">test_keycard_command_send_tx_eth_1_1_chat</a>, id: 702186 </b></td></tr><tr><td><p><b>Class: TestContactBlockMigrateKeycardMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/127b86273aee43e1ba5346611f788fbf?auth=61ca8fd37b367111dfddebe8ef8f5f29#133">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f857ee7768c84fb6abeefe2acdb8f8b3?auth=0a736679c4267cb69211ab5b19971947#172">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>64. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892644">test_wallet_tx_history_copy_tx_hash_on_cellular</a>, id: 700756 </b></td></tr><tr><td><p><b>Class: TestWalletManagementDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f41b5bca6c3f45208d5c5b2bb0ab5b2b?auth=fd9f99e71b120835b46835a337bcd92c#116">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>65. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892628">test_group_chat_add_new_member_activity_centre</a>, id: 700732 </b></td></tr><tr><td><p><b>Class: TestGroupChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/fe12464090874b3e931b720d1684deed?auth=86365435160d9c916e48bd9c52ca6c80#147">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/9958600248224e8b82d4ea0895cca118?auth=f774c6cba0e6284c1902c9850c6c6208#120">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/75bbbe8203d741fa9ec8702f9aca66b3?auth=04d70d7bf1b4aa4013b20ace990df098#78">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>66. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892665">test_browser_web3_permissions_testdapp</a>, id: 702078 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#576">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>67. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892674">test_restore_account_migrate_multiaccount_to_keycard_db_saved</a>, id: 702177 </b></td></tr><tr><td><p><b>Class: TestContactBlockMigrateKeycardMultipleSharedDevices</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/127b86273aee43e1ba5346611f788fbf?auth=61ca8fd37b367111dfddebe8ef8f5f29#554">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/f857ee7768c84fb6abeefe2acdb8f8b3?auth=0a736679c4267cb69211ab5b19971947#385">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>68. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892660">test_browser_connection_is_secure_not_secure_warning</a>, id: 702073 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#299">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>69. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892613">test_public_chat_fetch_more_history</a>, id: 5675 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#54">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>70. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892631">test_public_chat_link_send_open</a>, id: 700736 </b></td></tr><tr><td><p><b>Class: TestPublicChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f4fd24d3b9bd4cdba84d51440ecc3eb5?auth=62d5d0fddaf4087a0f8e831b9a23fcde#263">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/60f9247c10834c278be8564228c98d5b?auth=d55103e69af7a153dbb65a3451527815#191">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>71. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892616">test_1_1_chat_command_send_tx_eth_outgoing_tx_push</a>, id: 6253 </b></td></tr><tr><td><p><b>Class: TestCommandsMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/eceb4772387f42b0b085aba967b25cf0?auth=04ba18d8d3ff71837e69e9938b5ea326#88">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/de559aba346c48069881752c907a0c8c?auth=5adf98350d693108090df86c534ec14d#74">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>72. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892608">test_send_tx_sign_message_2tx_in_batch_tx_filters_request_stt_testdapp</a>, id: 5342 </b></td></tr><tr><td><p><b>Class: TestSendTxDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/76a39fc9243143538e19cae6114a3e9b?auth=87e0b0236541f1e45d3d45e6bd6b6703#299">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>73. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892626">test_public_chat_emoji_send_copy_paste_reply</a>, id: 700719 </b></td></tr><tr><td><p><b>Class: TestPublicChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f4fd24d3b9bd4cdba84d51440ecc3eb5?auth=62d5d0fddaf4087a0f8e831b9a23fcde#147">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/60f9247c10834c278be8564228c98d5b?auth=d55103e69af7a153dbb65a3451527815#90">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>74. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892651">test_send_tx_eth_check_logcat</a>, id: 700763 </b></td></tr><tr><td><p><b>Class: TestSendTxDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/76a39fc9243143538e19cae6114a3e9b?auth=87e0b0236541f1e45d3d45e6bd6b6703#222">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>75. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892619">test_public_chat_mark_all_messages_as_read</a>, id: 6270 </b></td></tr><tr><td><p><b>Class: TestPublicChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f4fd24d3b9bd4cdba84d51440ecc3eb5?auth=62d5d0fddaf4087a0f8e831b9a23fcde#317">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/60f9247c10834c278be8564228c98d5b?auth=d55103e69af7a153dbb65a3451527815#277">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>76. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892618">test_1_1_chat_command_decline_eth_push_changing_state</a>, id: 6265 </b></td></tr><tr><td><p><b>Class: TestCommandsMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/eceb4772387f42b0b085aba967b25cf0?auth=04ba18d8d3ff71837e69e9938b5ea326#169">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/de559aba346c48069881752c907a0c8c?auth=5adf98350d693108090df86c534ec14d#157">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>77. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892645">test_send_tx_set_recipient_options</a>, id: 700757 </b></td></tr><tr><td><p><b>Class: TestSendTxDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/76a39fc9243143538e19cae6114a3e9b?auth=87e0b0236541f1e45d3d45e6bd6b6703#564">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>78. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892663">test_browser_delete_close_tabs</a>, id: 702076 </b></td></tr><tr><td><p><b>Class: TestPublicChatBrowserOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/96d53560ea0b4f779d71aa6c4accb164?auth=bcfb30b46046398f36091fe90d7cf06b#409">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>79. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892683">test_pairing_sync_public_chat_add_remove</a>, id: 702199 </b></td></tr><tr><td><p><b>Class: TestPairingSyncMultipleDevicesMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f8d70ad9fc9c4d8bb2593edbd22afaa2?auth=cd8cb013e81a4cbf531036502f14b536#449">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/708540b1ae634f81a52d5d5d8c8bb1f6?auth=22cf2fd1d6a5ff81b9a821b9b10b1c7a#147">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>80. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892654">test_wallet_fetching_balance_after_offline_insufficient_funds_errors</a>, id: 700766 </b></td></tr><tr><td><p><b>Class: TestWalletManagementDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f41b5bca6c3f45208d5c5b2bb0ab5b2b?auth=fd9f99e71b120835b46835a337bcd92c#742">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>81. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892632">test_public_chat_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 700737 </b></td></tr><tr><td><p><b>Class: TestPublicChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f4fd24d3b9bd4cdba84d51440ecc3eb5?auth=62d5d0fddaf4087a0f8e831b9a23fcde#281">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/60f9247c10834c278be8564228c98d5b?auth=d55103e69af7a153dbb65a3451527815#214">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>82. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892646">test_wallet_manage_assets</a>, id: 700758 </b></td></tr><tr><td><p><b>Class: TestWalletManagementDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/f41b5bca6c3f45208d5c5b2bb0ab5b2b?auth=fd9f99e71b120835b46835a337bcd92c#256">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>83. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892600">test_group_chat_push_system_messages_when_invited</a>, id: 3994 </b></td></tr><tr><td><p><b>Class: TestGroupChatMultipleDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/fe12464090874b3e931b720d1684deed?auth=86365435160d9c916e48bd9c52ca6c80#124">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://saucelabs.com/jobs/9958600248224e8b82d4ea0895cca118?auth=f774c6cba0e6284c1902c9850c6c6208#77">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://saucelabs.com/jobs/75bbbe8203d741fa9ec8702f9aca66b3?auth=04d70d7bf1b4aa4013b20ace990df098#65">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>84. <a href="https://ethstatus.testrail.net/index.php?/tests/view/2892656">test_keycard_relogin_after_restore</a>, id: 700768 </b></td></tr><tr><td><p><b>Class: TestKeycardTxOneDeviceMerged</b></p><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://saucelabs.com/jobs/925aed991abc41c7a610d918d992a1be?auth=5e4eb8ef6d344f1849d6d949aa31411e#243">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

Parveshdhull committed:
Implement React Native Reanimated V2

Parveshdhull committed:
Switcher and Bottom Tabs Animations and UI Performance Improvements

- Migrated Switcher animations to Reanimated V2
- Added bottom tabs & Stacks Animations
- Improved bottom tabs, tab changing performance
- Polished android & IOS UI
